internal/discover/mounts_test.go

```diff
@@ -188,3 +188,131 @@ func TestMounts(t *testing.T) {
 		})
 	}
 }
+
+func TestMountsDeterministicOrder(t *testing.T) {
+	logger, _ := testlog.NewNullLogger()
+	mountOptions := []string{
+		"ro",
+		"nosuid",
+		"nodev",
+		"rbind",
+		"rprivate",
+	}
+
+	// Create a mock locator that returns paths in a non-deterministic order
+	mockLocator := &lookup.LocatorMock{
+		LocateFunc: func(s string) ([]string, error) {
+			// Return paths in different orders for each call
+			if s == "lib1" {
+				return []string{
+					"/usr/lib/libGLX_nvidia.so",
+					"/usr/lib/libGLX_indirect.so",
+					"/usr/lib/libnvidia.so",
+				}, nil
+			}
+			if s == "lib2" {
+				return []string{
+					"/usr/lib/libnvidia.so",
+					"/usr/lib/libGLX_indirect.so",
+					"/usr/lib/libGLX_nvidia.so",
+				}, nil
+			}
+			return []string{s}, nil
+		},
+	}
```

Comment:
The issue was not that the locator was returning things in a non-deterministic order, but that we were using a MAP to deduplicate the returned values. When we construct the slice from the map keys the order could change.

Reply:
Agree, I have simplified this to a single return.